### PR TITLE
Change coordinator to query and add `m3query` service

### PR DESCRIPTION
**Pull request by benraskin92**

### CODE REVIEW

File: docker/m3coordinator/Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ LABEL maintainer="The M3DB Authors <m3db@googlegroups.com>"
 EXPOSE 7201/tcp 7203/tcp
 
 COPY --from=builder /go/src/github.com/m3db/m3db/bin/m3coordinator /bin/
-COPY --from=builder /go/src/github.com/m3db/m3db/src/coordinator/config/m3coordinator-local-etcd.yml /etc/m3coordinator/m3coordinator.yml
+COPY --from=builder /go/src/github.com/m3db/m3db/src/query/config/m3coordinator-local-etcd.yml /etc/m3coordinator/m3coordinator.yml
```
Comment (prateek) [inline docker/m3coordinator/Dockerfile]: hm do you want to chance the directory name here too

File: scripts/process-cover.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ TARGETS=("dbnode" "coordinator" "m3ninx")
 target_patterns() {
     case $1 in
         'dbnode') echo "^mode|github.com/m3db/m3db/src/dbnode|github.com/m3db/m3db/src/cmd/services/m3dbnode";;
-        'coordinator') echo "^mode|github.com/m3db/m3db/src/coordinator|github.com/m3db/m3db/src/cmd/services/m3coordinator";;
+        'coordinator') echo "^mode|github.com/m3db/m3db/src/query|github.com/m3db/m3db/src/cmd/services/m3coordinator";;
```
Comment (prateek) [inline scripts/process-cover.sh]: hmm maybe rename this to 'query' too

Comment (prateek) [reply scripts/process-cover.sh]: similarly for the `.codecov.yml` file and -F flags in the makefile

File: docker/m3coordinator/Dockerfile (hunk 1 above)
Comment (robskillington) [reply docker/m3coordinator/Dockerfile]: Yeah maybe we should move these config files outside of `src` completely?  It's weird we have some YAML files in these dirs floating around.

File: src/cmd/services/m3coordinator/main/main.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ import (
 	_ "net/http/pprof" // pprof: for debug listen server if configured
 	"os"
 
-	"github.com/m3db/m3db/src/coordinator/services/m3coordinator/server"
+	"github.com/m3db/m3db/src/query/services/m3coordinator/server"
```
Comment (robskillington) [inline src/cmd/services/m3coordinator/main/main.go]: The plan is to follow up and move the package being imported here to `src/coordinator/server/...` yeah?

File: docker/m3coordinator/Dockerfile (hunk 1 above)
Comment (benraskin92) [reply docker/m3coordinator/Dockerfile]: i guess we should figure out if we want to have an m3coordinator and an m3query dockerfile? we can discuss today once everyone is in

**Pull request merged**